src/compas/geometry/shapes/box.py

```diff
@@ -744,3 +744,52 @@ def contains_points(self, points, tol=1e-6):
             results[index] = True
 
         return results
+
+    def has_collision(self, other):
```

Comment:
I think this method should be called something with `intersection_...()`, maybe `intersection_with_box()`, to match other existing functions/methods in the `compas.geometry` module and in the surfaces stuff.

Reply:
ahh ok! that's a good point, I'm not sure what could be a good name then.. `is_intersected_by_box(other)`? that's weird and too verbose, isn't it? `is_colliding_with_box(other)`?

/cc @tomvanmele @chenkasirer @Licini @romanarust suggestions?